lib/Entity/Schedule.php

```diff
@@ -474,6 +474,37 @@ public function validate()
         // Check recurrenceDetail every is positive
         if ($this->recurrenceType != '' && ($this->recurrenceDetail === null || $this->recurrenceDetail <= 0))
             throw new InvalidArgumentException(__('Repeat every must be a positive number'), 'recurrenceDetail');
+        
+        // Check and disallow Minute/Hourly repeats to be indefinite (or too long)
+        $twelveHoursInSeconds = 12 * 60 * 60;
+        $oneWeekInSeconds = 24 * 7 * 60 * 60;
+        if($this->recurrenceType == 'Minute') {
+
+            if (empty($this->recurrenceRange)) {
+                throw new InvalidArgumentException(__('You must provide a date and time to end the Repeat'), 'recurrenceRange');
```

Comment:
Can we make this a little friendlier? Something like:

"An end time is needed for an event that repeats every minute."